gtest/src/program.rs

```diff
@@ -440,28 +440,24 @@ impl<'a> Program<'a> {
         self.id
     }
 
-    pub fn meta_state<E: Encode, D: Decode>(&self, payload: E) -> Result<D> {
-        D::decode(&mut self.meta_state_with_bytes(payload.encode())?.as_slice()).map_err(Into::into)
+    pub fn read_state(&self) -> Result<Vec<u8>> {
+        self.manager.borrow_mut().read_state(&self.id)
     }
 
-    pub fn meta_state_with_bytes(&self, payload: impl AsRef<[u8]>) -> Result<Vec<u8>> {
-        self.manager.borrow_mut().call_meta(
+    pub fn read_state_with_map(
+        &self,
+        mapping_wasm: Vec<u8>,
+        mapping_wasm_function_name: impl Into<String>,
```

Comment:
IMO suboptimal. Why if we want to pass `&str`?

Reply:
As I see it makes sense to use `AsRef<str>` or similar only if we would implement the `WasmEntry` trait for it. Not crucial for the testing crate, though.

Reply:
replaced by using &str. Found here https://www.philipdaniels.com/blog/2019/rust-api-design/#:~:text=Into%20has%20a%20similar%20effect,String%3E%20%7B%20...%20%7D that is more idioamatic for rust to use &str instead of AsRef<str>

Reply:
AsRef<str> is more general than &str. You can find more use case details in the "Rust for Rustaceans" book.

![image](https://user-images.githubusercontent.com/66209982/216376645-d0b90223-b7c3-4442-916a-caae124df762.png)